packages/apollo-server-cache-redis-cluster/src/index.ts

```diff
@@ -0,0 +1,72 @@
+import { KeyValueCache } from 'apollo-server-caching';
+import Redis, { ClusterOptions, ClusterNode, RedisOptions } from 'ioredis';
+import { promisify } from 'util';
+import DataLoader from 'dataloader';
+
+export class RedisCache implements KeyValueCache {
+  protected client: any;
+  readonly defaultSetOptions = {
+    ttl: 300,
+  };
+
+  protected loader: DataLoader<string, string>;
+
+  constructor(options?: RedisOptions) {
+    const client = new Redis(options);
+    client.mget = promisify(client.mget).bind(client);
+    client.set = promisify(client.set).bind(client);
+    client.flushdb = promisify(client.flushdb).bind(client);
+    client.quit = promisify(client.quit).bind(client);
+
+    this.client = client;
+
+    this.loader = new DataLoader(keys => this.client.mget(keys), {
+      cache: false,
+    });
+  }
+
+  async set(
+    key: string,
+    data: string,
+    options?: { ttl?: number },
+  ): Promise<void> {
+    const { ttl } = Object.assign({}, this.defaultSetOptions, options);
+    await this.client.set(key, data, 'EX', ttl);
+  }
+
+  async get(key: string): Promise<string | undefined> {
+    const reply = await this.loader.load(key);
+    // reply is null if key is not found
+    if (reply !== null) {
+      return reply;
+    }
+    return;
+  }
+
+  async flush(): Promise<void> {
+    await this.client.flushdb();
+  }
+
+  async close(): Promise<void> {
+    await this.client.quit();
+    return;
+  }
+}
+
+export class RedisClusterCache extends RedisCache {
+  constructor(nodes: ClusterNode[], options?: ClusterOptions) {
+    super();
+    const client = new Redis.Cluster(nodes, options);
+
+    client.mget = promisify(client.mget).bind(client);
+    client.set = promisify(client.set).bind(client);
+    client.flushdb = promisify(client.flushdb).bind(client);
+    client.quit = promisify(client.quit).bind(client);
+
+    this.client = client;
+
+    this.loader = new DataLoader(keys => this.client.mget(keys), {
+      cache: false,
+    });
```

Comment:
I think using DataLoader for distinct keys is good, but redis cluster doesn't support crossslot mget. So maybe it will be something like this?
```typescript
    this.loader = new DataLoader(
      keys => Promise.all(keys.map(key => this.client.get(key).catch(error => error))),
      { cache: false }
    );
```